java/algorithm/src/main/java/org/apache/arrow/algorithm/sort/DefaultVectorComparators.java

```diff
@@ -44,6 +85,84 @@ public int compareNotNull(int index1, int index2) {
     }
   }
 
+  /**
+   * Default comparator for long integers.
+   * The comparison is based on values, with null comes first.
+   */
+  public static class LongComparator extends VectorValueComparator<BigIntVector> {
+
+    public LongComparator() {
+      super(Long.SIZE / 8);
+    }
+
+    @Override
+    public int compareNotNull(int index1, int index2) {
+      long value1 = vector1.get(index1);
+      long value2 = vector2.get(index2);
+
+      long result = value1 - value2;
```

Comment:
just return result?

Reply:
I guess this protects against overflow?

Reply:
You are right. This is to protect against overflow which may happen when converting a long to an int.